File: DevExpress VCL 23.2.6 Full Source/VCL/ExpressGantt Control/Demos/CBuilder/ExtendedAttributesDemo/ExtendedAttributesDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ExtendedAttributesDemoMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxDropDownEdit"
#pragma link "cxEdit"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma link "cxMaskEdit"
#pragma link "cxSpinEdit"
#pragma link "cxTextEdit"
#pragma link "cxButtons"
#pragma link "cxGroupBox"
#pragma link "dxCore"
#pragma link "dxGanttControl"
#pragma link "dxGanttControlAssignments"
#pragma link "dxGanttControlCustomClasses"
#pragma link "dxGanttControlCustomSheet"
#pragma link "dxGanttControlResources"
#pragma link "dxGanttControlTasks"
#pragma link "dxGanttControlViewChart"
#pragma link "dxGanttControlViewResourceSheet"
#pragma link "dxGanttControlViewTimeline"
#pragma link "dxGanttControlExtendedAttributes"
#pragma link "dxLayoutContainer"
#pragma link "dxLayoutControl"
#pragma link "dxLayoutControlAdapters"
#pragma link "dxLayoutcxEditAdapters"
#pragma link "dxLayoutLookAndFeels"
#pragma link "dxGanttControlExtendedAttributes"
#pragma resource "*.dfm"
TExtendedAttributesDemoMainForm *ExtendedAttributesDemoMainForm;
//---------------------------------------------------------------------------
__fastcall TExtendedAttributesDemoMainForm::TExtendedAttributesDemoMainForm(TComponent* Owner)
	: TDemoBasicMainForm(Owner)
{
  GanttControl->DataModel->LoadFromFile("..\\..\\Data\\ExtendedAttributesDemo.xml");
}
//---------------------------------------------------------------------------
void __fastcall TExtendedAttributesDemoMainForm::GanttControlDataModelLoaded(TObject *Sender)
{
  CreateExtendedAttributeColumns(GanttControl->ViewChart->OptionsSheet);
  GanttControl->ViewChart->OptionsSheet->Columns->Items[0]->Visible = false;
  GanttControl->ViewChart->OptionsSheet->Columns->Items[1]->Visible = false;
  GanttControl->ViewChart->OptionsSheet->Columns->Items[3]->Visible = false;
  GanttControl->ViewChart->OptionsSheet->Columns->Items[7]->Visible = false;
  CreateExtendedAttributeColumns(GanttControl->ViewResourceSheet->OptionsSheet);
}
//---------------------------------------------------------------------------

void __fastcall TExtendedAttributesDemoMainForm::lcbAlwaysShowEditorClick(TObject *Sender)

{
	GanttControl->OptionsBehavior->AlwaysShowEditor = lcbAlwaysShowEditor->Checked;
}
//---------------------------------------------------------------------------
void __fastcall TExtendedAttributesDemoMainForm::CreateExtendedAttributeColumns(TdxGanttControlViewSheetOptions *AOptions)
{
    AOptions->Columns->Reset();
    AOptions->RetrieveMissingExtendedAttributeColumns();
    AOptions->AddExtendedAttributeColumns();

}
//---------------------------------------------------------------------------

File: DevExpress VCL 23.2.6 Full Source/VCL/ExpressGantt Control/Demos/Delphi/ExtendedAttributesDemo/ExtendedAttributesDemoMain.pas
unit ExtendedAttributesDemoMain;

interface

{$I cxVer.inc}

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, Menus, ComCtrls, StdCtrls,
  dxCore, cxContainer, cxEdit, cxTextEdit, cxDropDownEdit, cxMaskEdit, cxSpinEdit,
  cxGraphics, cxClasses, cxControls,
{$IFDEF EXPRESSSKINS}
  dxSkinsdxRibbonPainter,
  dxSkinsdxBarPainter,
{$ENDIF}
  cxLookAndFeels, cxLookAndFeelPainters, dxLayoutControl,
  dxLayoutContainer, dxLayoutLookAndFeels, dxLayoutcxEditAdapters, DemoBasicMain,
  dxGanttControlCustomClasses, dxGanttControl, dxGanttControlCustomSheet, dxGanttControlViewChart,
  dxGanttControlViewResourceSheet, dxGanttControlViewTimeline, dxGanttControlTasks, dxGanttControlAssignments,
  dxGanttControlResources, cxGroupBox, dxLayoutControlAdapters, cxButtons, dxGanttControlExtendedAttributes,
  ActnList, ImgList, cxImageList;

type
  TExtendedAttributesDemoMainForm = class(TDemoBasicMainForm)
    lcbAlwaysShowEditor: TdxLayoutCheckBoxItem;
    procedure GanttControlDataModelLoaded(Sender: TObject);
    procedure lcbAlwaysShowEditorClick(Sender: TObject);
  public
    constructor Create(AOwner: TComponent); override;
  end;

var
  ExtendedAttributesDemoMainForm: TExtendedAttributesDemoMainForm;

implementation

{$R *.dfm}

uses
  dxGanttControlCustomView;

constructor TExtendedAttributesDemoMainForm.Create(AOwner: TComponent);
begin
  inherited Create(AOwner);
  GanttControl.DataModel.LoadFromFile('..\..\Data\ExtendedAttributesDemo.xml');
end;

procedure TExtendedAttributesDemoMainForm.GanttControlDataModelLoaded(
  Sender: TObject);

  procedure CreateExtendedAttributeColumns(AOptions: TdxGanttControlViewSheetOptions);
  begin
    AOptions.Columns.Reset;
    AOptions.RetrieveMissingExtendedAttributeColumns;
    AOptions.AddExtendedAttributeColumns;
  end;

begin
  inherited;
  CreateExtendedAttributeColumns(GanttControl.ViewChart.OptionsSheet);
  GanttControl.ViewChart.OptionsSheet.Columns[0].Visible := False;
  GanttControl.ViewChart.OptionsSheet.Columns[1].Visible := False;
  GanttControl.ViewChart.OptionsSheet.Columns[3].Visible := False;
  GanttControl.ViewChart.OptionsSheet.Columns[7].Visible := False;
  CreateExtendedAttributeColumns(GanttControl.ViewResourceSheet.OptionsSheet);
end;

procedure TExtendedAttributesDemoMainForm.lcbAlwaysShowEditorClick(
  Sender: TObject);
begin
  inherited;
  GanttControl.OptionsBehavior.AlwaysShowEditor := lcbAlwaysShowEditor.Checked;
end;

end.
